li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/utils/LiKafkaClientsUtils.java

```diff
@@ -101,4 +106,25 @@ public static void dumpStacksForAllLiveThreads(Set<Thread> threads) {
       }
     }
   }
+
+  public static Map<String, String> toStringMap(Properties props, List<String> errors) {
+    if (props == null) {
+      return null;
+    }
+    if (props.isEmpty()) {
+      return Collections.emptyMap();
+    }
+    Map<String, String> translated = new HashMap<>(props.size());
+    props.forEach((k, v) -> {
+      String sk = k == null ? null : k.toString();
+      String sv = v == null ? null : v.toString();
+      if (translated.containsKey(sk) && errors != null) {
```

Comment:
your code is only equivalent if null values cannot exist (put can return null because null was the previous value under a key, not just because the key was not found).
however, after navina's comment above (Properties cannot have nulls at all) i can safely switch